decentraland/unity-renderer
# feat: profanity chat filtering

## What does this PR change?

`Closes #1442 `

- Badwords in general chat are replaced with `***`
- Added new toggle setting to activate/deactivate this feature in general settings menu
- Private messages are not filtered
- General/Quality/Audio settings persistence has been changed to support default values when the user already has a saved setting

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=feat/profanity-chat-filtering
2. Go to settings menu, at general tab, you should have "Profanity Filter" toggle enabled.
3. Write any badword like "shit im filtered" in the general chat. You should see the message in the chat like this: "**** im filtered"
4. Write any private message to a friend that contains a badword like "im thinking with my ass". You should see the message unfiltered (private chat window and/or general chat)
5. Go to settings menu, at general tab, disable "Profanity Filter" toggle.
6. Repeat step 3 and 4. None of those messages should be filtered.
7. To to settings menu, tweak some general/quality/audio settings. Restart app, check changes are correctly saved. Then hit "Reset All" button. All changed settings should be restored

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
